src/linkify-element.js

```diff
@@ -32,8 +32,11 @@ function tokensToNodes(tokens, opts, doc) {
 
 	for (let i = 0; i < tokens.length; i++) {
 		let token = tokens[i];
+		let validated = options.resolve(opts.validate,
+			token.hasProtocol ? token.hasProtocol() : false,
```

Comment:
In that sense, I wouldn't be opposed to passing the whole token as the final argument!